src/layer-composer/generators/basemap/basemap-layers.ts

```diff
@@ -1,27 +1,73 @@
-import { Group } from '../../../types'
+import { Group, Dictionary } from '../../../types'
+import { Layer, AnySourceImpl } from 'mapbox-gl'
 
-const layers = {
-  'north-star': {
-    source: {
-      tiles: [
-        'https://mt1.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
-        'https://mt2.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
-        'https://mt3.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
-      ],
-      type: 'raster',
-      tileSize: 256,
+export const BASEMAPS = {
+  Satellite: 'satellite',
+  Landmass: 'landmass',
+  Graticules: 'graticules',
+}
+
+const BASEMAP_VECTOR_SOURCE = 'basemap_vector'
+
+export const layers: Dictionary<Layer> = {
+  [BASEMAPS.Satellite]: {
+    type: 'raster',
+    id: BASEMAPS.Satellite,
+    source: BASEMAPS.Satellite,
+    metadata: {
+      group: Group.Basemap,
+    },
+  },
+  [BASEMAPS.Landmass]: {
+    type: 'fill',
+    id: BASEMAPS.Landmass,
+    source: BASEMAP_VECTOR_SOURCE,
+    'source-layer': 'landmass',
+    metadata: {
+      group: Group.BasemapFill,
+    },
+    paint: {
+      'fill-color': '#374A6D',
+      'fill-outline-color': 'white',
+      'fill-opacity': 0.99,
+    },
+  },
+  [BASEMAPS.Graticules]: {
+    type: 'line',
+    id: BASEMAPS.Graticules,
+    source: BASEMAP_VECTOR_SOURCE,
+    'source-layer': 'graticules',
+    metadata: {
+      group: Group.BasemapForeground,
     },
-    layers: [
-      {
-        id: 'north-star',
-        type: 'raster',
-        source: 'north-star',
-        metadata: {
-          group: Group.Basemap,
-        },
+    paint: {
+      'line-color': '#ffffff',
+      'line-opacity': {
+        base: 1,
+        stops: [
+          [0, 0.7],
+          [8, 0.7],
+          [9, 0.2],
+        ],
       },
-    ],
+      'line-width': 0.5,
+    },
   },
 }
 
-export default layers
+export const sources: Dictionary<AnySourceImpl> = {
+  [BASEMAPS.Satellite]: {
+    tiles: [
+      'https://mt1.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
+      'https://mt2.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
+      'https://mt3.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
+    ],
+    type: 'raster',
+    tileSize: 256,
+  },
+  [BASEMAP_VECTOR_SOURCE]: {
+    type: 'vector',
+    tiles: ['http://localhost:9090/tiles-basemap/{z}/{x}/{y}.pbf'],
```

Comment:
Would be nice to have a `tiles` params in the basemapGeneratorConfig to be able to resuse this

Reply:
I'm not sure. If we want this we might prefer having a more generic `VectorTilesGenerator` or something. I think the Basemap is a thing we want to drop in any project and forget about it, not something that should be configurable.
Let's reasses when we have a use case for this.